psl-core/src/main/java/org/linqs/psl/model/rule/FakeRule.java

```diff
@@ -93,11 +93,6 @@ public void ground(Constant[] constants, Map<Variable, Integer> variableMap, Ato
         // Pass.
     }
 
-    @Override
-    public int hashCode() {
-        throw new UnsupportedOperationException("FakeRules should not be stored in hashing structures.");
-    }
-
```

Comment:
Add a class comment that briefly talks about the semantics of fakes rules (e.g. hash and equality semantics).